elastic/kibana
# Replace the KQL filter with the Unified Search Bar in the custom threshold rule

Closes #165195, partially implements #163819

## Summary

This PR replaces the KQL filter with the Unified Search Bar in the custom threshold rule. Also, it saves the query in the searchConfiguration instead of a separate field.

<img src="https://github.com/elastic/kibana/assets/12370520/98609c52-03ef-4e87-9cd5-4baf4c91d362" width=500 />

Also, it shows a message that the filter query is invalid:

<img src="https://github.com/elastic/kibana/assets/12370520/6253e9d9-8d5f-4ca5-adf1-41e8469865f6" width=500 />


## 🧪 How to test
- Create a custom threshold rule and add a filter
- Make sure the filter is applied correctly on the generated alerts
- Check the saved object for the rule, you should not see the filterQuery field anymore, and instead, the query should be in the searchConfiguration field
- Edit the rule and add an invalid input, you should see an error message, and you should not be able to save the rule
